library/std/src/error.rs

```diff
@@ -871,6 +857,20 @@ impl Error for alloc::ffi::IntoStringError {
     }
 }
 
+impl<'a> dyn Error + 'a {
+    /// Request a reference of type `T` as context about this error.
+    #[unstable(feature = "error_generic_member_access", issue = "99301")]
+    pub fn request_ref<T: ?Sized + 'static>(&'a self) -> Option<&'a T> {
+        core::any::request_ref(self)
+    }
+
+    /// Request a value of type `T` as context about this error.
+    #[unstable(feature = "error_generic_member_access", issue = "99301")]
+    pub fn request_value<T: 'static>(&'a self) -> Option<T> {
+        core::any::request_value(self)
+    }
+}
```

Comment:
I had to change this from `'static` to make it work with `std::error::Report` which already supports non `'static` errors and tests reporting borrowed errors.